src/sentry/static/sentry/app/utils/performance/quickTrace/utils.ts

```diff
@@ -1,5 +1,236 @@
+  const events: TraceFull[] = [];
+
+  /**
+   * First find a path from the root transaction to the current transaction via
+   * a breadth first search. This adds all transactions from the root to the
+   * current transaction (excluding the current transaction itself), to the
+   * relevant path.
+   */
+  const paths: PathNode[] = [{event: traceFull, path: []}];
+  while (paths.length) {
+    const current = paths.shift()!;
+    if (current.event.event_id === currentEvent.id) {
+      for (const node of current.path) {
+        relevantPath.push(node);
+      }
+      events.push(current.event);
+    } else {
+      const path = [...current.path, simplifyEvent(current.event)];
+      for (const child of current.event.children) {
+        paths.push({event: child, path});
+      }
+    }
+  }
+
+  /**
+   * Traverse all transactions from current transaction onwards and add
+   * them all to the relevant path.
+   */
+  while (events.length) {
+    const current = events.shift()!;
+    for (const child of current.children) {
+      events.push(child);
+    }
+    relevantPath.push(simplifyEvent(current));
+  }
+
+  return relevantPath;
+}
+
+function simplifyEvent(event: TraceFull): EventLite {
+  return omit(event, 'children');
+}
+
+type ParsedQuickTrace = {
+  /**
+   * `null` represents the lack of a root. It may still have a parent
+   */
+  root: EventLite | null;
+  /**
+   * `[]` represents the lack of ancestors in a full quick trace
+   * `null` represents the uncertainty of ancestors in a lite quick trace
+   */
+  ancestors: TraceLite | null;
+  /**
+   * `null` represents either the lack of a direct parent or the uncertainty
+   * of what the parent is
+   */
+  parent: EventLite | null;
+  current: EventLite;
+  /**
+   * `[]` represents the lack of children in a full/lite quick trace
+   */
+  children: TraceLite;
+  /**
+   * `[]` represents the lack of descendants in a full quick trace
+   * `null` represents the uncertainty of descendants in a lite quick trace
+   */
+  descendants: TraceLite | null;
+};
+
+export function parseQuickTrace(
+  traceType: QuickTraceQueryChildrenProps['type'],
+  trace: TraceLite,
+  event: Event
+): ParsedQuickTrace | null {
+  if (traceType === 'empty') {
+    return null;
+  }
+
+  const isFullTrace = traceType === 'full';
+
+  const current = trace.find(e => e.event_id === event.id) ?? null;
+  if (current === null) {
+    /**
+     * The current event should always be present in the trace, if not
+     * there is no reason to at the rest for the quick trace.
+     */
+    Sentry.captureException(new Error('Current event not in quick trace'));
+    return null;
+  }
+
+  /**
+   * The parent event is the direct ancestor of the current event.
+   * This takes priority over the root, meaning if the parent is
+   * the root of the trace, this favours showing it as the parent.
+   */
+  const parent = trace.find(e => e.event_id === current.parent_event_id) ?? null;
```

Comment:
```suggestion
  const parent = current.parent_event_id ? (trace.find(e => e.event_id === current.parent_event_id) ?? null) : null;
```

Reply:
Is this extra check necessary or just a optimization?

Reply:
optimization.